### DISCUSSION TITLE
Layer Script for Special Words in Content of a Page?

### DISCUSSION THREAD

**Discussion by bashuss** [OrchardCMS/OrchardCore#15510] [Q&A]
Is there a way to create a layer for placing a widget, that finds specific keywords in the content of a page or in meta keywords?

This would be helpful to automatically add Trademark notes, without the need to do that as part of the content.

1. I guess, this would need a custom javascript method defined in a module? Or can I use javascript like "document.body ..." to achieve this. 
2. Layer rule expects a boolean as result of the javascript. May I code like the body of a function and end with `return result;`, or can I just use an expression as it would be used inside `if ( expression ) { ... }`?
3. I still don't really understand how far we can go with the scripting engine and how much that differs in different context (workflow, layers, ...). I'd love to see documentation, with examples which show how far you could go.

**Comment by Piedone** [comment]
There's no feature for this, and you'd need to implement a custom rule (or extend Jint for a custom JS function or something) that fetches the text of the current content item. Note that this JS runs on the server-side, so you can't access the `document`.
